smf/parser.js

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env node
+
+// ##EZPAARSE
+
+/*jslint maxlen: 150*/
+'use strict';
+var Parser = require('../.lib/parser.js');
+
+/**
+ * Recognizes the accesses to the platform Société Mathématique de France
+ * @param  {Object} parsedUrl an object representing the URL to analyze
+ *                            main attributes: pathname, query, hostname
+ * @param  {Object} ec        an object representing the EC whose URL is being analyzed
+ * @return {Object} the result
+ */
+module.exports = new Parser(function analyseEC(parsedUrl, ec) {
+  var result = {};
+  var path   = parsedUrl.pathname;
+  // uncomment this line if you need parameters
+  // var param  = parsedUrl.query || {};
+
+  // use console.error for debuging
+  // console.error(parsedUrl);
+
+  var match;
+
+  if ((match = /^(\/en)?\/[a-z]+\/(\/)?([a-z]+)\/([0-9]+)(\/([0-9]+))?\/(pdf|html)\/([a-z\_\-]+([0-9]+)\_([^.]+))\.(pdf|php)$/i.exec(path)) !== null) {
```

Comment:
`\/(\/)?` involves a useless capturing group and can be replaced with something simpler, like `\/\/?` or `\/{1,2}` or `\/+`.
`([a-z\_\-]+([0-9]+)\_([^.]+))` involves two capturing groups that aren't used. Try not to put too much capturing groups when you don't need them. (you can also use non-capturing groups `(?: ... )`)